docs/decisions/07-posts-data-model.md

```diff
@@ -0,0 +1,162 @@
+
+```mermaid
+erDiagram
+    Note zero or one -- one Post : posted
+    Image 0+ -- one Post : posted
+    Etc 0+ -- one Post : posted
+    Post 0+ -- 1+ Profile : created-by
+    Post 0+ -- 0+ Profile : liked-by
+    Post 0+ -- 0+ Profile : shared-by
+    Post 0+ -- 0+ Profile: addressed-to
+    Post 0+ -- 0+ Audience : visible-to
+    Post 0+ -- zero or one Post : in-reply-to
+    Profile 1+ -- 1+ Audience : member-of
+    
+    Post {
+        uuid id PK
+        uri uri "indexed"
+        uri thread "indexed"
+        timestamp created
+        timestamp updated
+        string client
+    }
+    
+    Note {
+        uuid id PK
+        uri uri "indexed"
+        string summary
+        string preview
+        string source
+    }
+    
+    Image {
+        uuid id PK
+        uri uri "indexed"
+        string summary
+        string preview
+        string source
+    }
+    
+    Etc {
+        uuid id PK
+        uri uri "indexed"
+        string summary
+        string preview
+        string source
+ }
+    
+```
+
+So, everything becomes a post, with some child content items. It doesn't currently happen (much?) in the wild, but this would allow us to easily handle activities that include multiple content-like objects (multiple notes, images, questions, mix-and-match, etc). The types above are mostly illustrative, not complete or final. The idea is that they would implement an interface, call it `IContent`, which would provide enough information to build a feed of posts. That would make it a lot easier to actually provide those feeds to clients. Without *something* like that, clients would have had to make at least one follow-up request to get the actual content of feed items that were otherwise only a list of IDs/URLs.
+
+```mermaid
+classDiagram
+    direction LR
+    class IContent {
+        <<interface>>
+        Uuid Id
+        Uri Uri
+        Uri Thread
+        string? Summary
+        string? Preview
+        string? Source
+        
+        Type() string
+    }
+    
+    class Post {
+        Uuid Id
+        Uri Uri
+        Uri Thread
+        DateTime Created
+        DateTime Updated
+        IList~IContent~ Contents
+        string Client
+    }
+    
+    class Note {
+        string Content
+        Type() "Note"
+    }
+
+    class Image {
+        MediaType MediaType
+        Type() "Image"
+    }
+    
+    class FeedDto {
+        Post Post
+        Profile CreatedBy
+        Profile SharedBy
+        Post InReplyTo
+        Uri Thread
+        Content Content
+    }
+    
+    class Content {
+        Uuid Id
+        Uri Uri
+        Uri Thread
+        string? Summary
+        string? Preview
+        string? Source
+        
+        Type()*
+    }
+
+    Content <|-- Note : inherits
+    Content <|-- Image : inherits
+    Content <|-- Etc : inherits
+    IContent <|-- Content : implements
+    Post --* IContent : contains
+    Post --> FeedDto : maps-to
+    FeedDto --* Content : contains
```

Comment:
I'm curious how would you see Quotes or Citation posts fitting into this model? Those types of content seem to lean into the graphical representation this wants to move away from.

Reply:
That's a fair point. I wish it was easier to find info about how other software federates quote posts. What I don't want to do is the mastodon thing of just having the one kind of post, which is basically a note with media attachments. So, I was kind of putting that off a little. The best thing might be if `Post` also implements `IContent`, so that the content list can also contain other posts.

Other options I can think of:
1. Note and Post are the same concept, and post gets a `public Post Quote` field.
2. Add a `public Post Quote` to the Note type, and future types that could meaningfully embed a quote.